tests/matrix_free/point_evaluation_21.cc

```diff
@@ -0,0 +1,158 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Like point_evaluation_01.cc but making sure we can copy and move
+// FEPointEvaluation objects.
+
+#include <deal.II/base/function_lib.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/matrix_free/fe_point_evaluation.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+#include <iostream>
+
+#include "../tests.h"
+
+
+
+template <int dim, typename Number = double>
+void
+test(const unsigned int degree)
+{
+  using namespace dealii;
+  Triangulation<dim> tria;
+
+  if (dim > 1)
+    GridGenerator::hyper_shell(tria, Point<dim>(), 0.5, 1, 6);
+  else
+    GridGenerator::subdivided_hyper_cube(tria, 2, 0, 1);
+
+  MappingQ<dim> mapping(degree);
+  deallog << "Mapping of degree " << degree << std::endl;
+
+  std::vector<Point<dim>> unit_points;
+  for (unsigned int i = 0; i < 13; ++i)
+    {
+      Point<dim> p;
+      for (unsigned int d = 0; d < dim; ++d)
+        p[d] = static_cast<Number>(i) / 17. + 0.015625 * d;
+      unit_points.push_back(p);
+    }
+
+  FE_Q<dim>     fe(degree);
+  FEValues<dim> fe_values(mapping,
+                          fe,
+                          Quadrature<dim>(unit_points),
+                          update_values | update_gradients);
+
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(fe);
+  Vector<Number> vector(dof_handler.n_dofs());
+
+  FEPointEvaluation<1, dim, dim, Number> evaluator(
+    mapping, fe, update_values | update_gradients);
+
+  FEPointEvaluation<1, dim, dim, Number> evaluator_copy(evaluator);
+
+  FEPointEvaluation<1, dim, dim, Number> evaluator_move(
+    std::move(evaluator_copy));
```

Comment:
In the code below, you are only testing the original object `evaluator`, but it would be nice to also test the `evaluator_move` object.